### PR TITLE
Restructure package (Alternative to #39)

### PR DESCRIPTION
**Pull request by azeey**
# 🎉 New feature

## Summary

After testing #39, I'm more convinced that it would be easier to put everything inside one python package. Last time I tried #39, I think I was having issues with imports.

The proposed package layout looks like:
```
.
└── sdformat_mjcf
    ├── __init__.py
    ├── mjcf_to_sdformat
    │   └── __init__.py
    ├── sdformat_to_mjcf
    │   └── __init__.py
    └── utils

```
Where we have `sdformat_mjcf` as the top level package. For the directory structure for for the distribution package (gz-mujoco/sdformat_mjcf), I opted to use the `src` layout as encouraged in the [pytest good practices](https://docs.pytest.org/en/7.1.x/explanation/goodpractices.html) page and explained in [this](https://blog.ionelmc.ro/2014/05/25/python-packaging/) blog. The advantage of the `src` layout is that our tests always test what's installed, not what's in the `src` directory.

In addition to shuffling files and fixing imports, I made a change to how `TEST_RESOURCE_DIR` is computed. This was necessary because the way we had it before, `__file__` was being evaluated in the context of the importing module and so, was dependent on the location of the importing module. The intended behavior was that `__file__` would always evaluate to the location of the `helper.py` module.

## Test it
All existing tests should pass.

## Checklist
- [ ] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: sdformat_mjcf/tests/sdformat_to_mjcf/test_add_material.py [hunk 1]
```diff
@@ -68,9 +69,9 @@ def test_material_pbr_bad_extension(self):
         pbr = sdf.Pbr()
         workflow = sdf.PbrWorkflow()
         workflow.set_type(sdf.PbrWorkflowType.METAL)
-        workflow.set_albedo_map(os.path.join(
-                                os.path.dirname(os.path.abspath(__file__)),
-                                "resources/box_obj/textures/albedo_map"))
+        workflow.set_albedo_map(
+            os.path.join(os.path.dirname(os.path.abspath(__file__)),
+                         "resources/box_obj/textures/albedo_map"))
```
Comment (scpeters) [inline sdformat_mjcf/tests/sdformat_to_mjcf/test_add_material.py:74]:
```suggestion
                         "resources", "box_obj", "textures", "albedo_map"))
```

**Review by scpeters — COMMENTED** [review]
I think the installation instructions in the README need to be updated, and the example usage of `mjcf2sdformat` should be copied from `mjcf_to_sdformat/README.md` to `sdformat_mjcf/README.md`

File: sdformat_mjcf/tests/sdformat_to_mjcf/test_add_material.py (hunk 1 above)
Comment (azeey) [reply sdformat_mjcf/tests/sdformat_to_mjcf/test_add_material.py]: Done in 96ab8c4.

**Review by azeey — COMMENTED** [review]
> I think the installation instructions in the README need to be updated, and the example usage of `mjcf2sdformat` should be copied from `mjcf_to_sdformat/README.md` to `sdformat_mjcf/README.md`

Done in 9229313. Also fixed some typos.

File: sdformat_mjcf/README.md [hunk 2]
```diff
@@ -59,3 +58,16 @@ extract the contents and run
...
 <path/to/mujoco>/bin/simulate new_file.xml
 ```
+
+To convert an SDFormat file to mjcf:
+
+```
+mjcf2sdformat path/to/file.xml new_file.sdf
+```
+
+To run the sdf file in GazeboSim, follow [this instructions to install Gazebo Sim](https://gazebosim.org/docs/latest/install)
```
Comment (scpeters) [inline sdformat_mjcf/README.md:68]:
```suggestion
To run the sdf file in GazeboSim, follow [these instructions to install Gazebo Sim](https://gazebosim.org/docs/latest/install)
```

**Review by scpeters — APPROVED** [review]
I think there are just two small typos to fix and then it can be merged

File: sdformat_mjcf/README.md [hunk 3]
```diff
@@ -59,3 +58,16 @@ extract the contents and run
 ```
 <path/to/mujoco>/bin/simulate new_file.xml
 ```
+
+To convert an SDFormat file to mjcf:
```
[the remark this answers is not in the document]
Comment (azeey) [reply sdformat_mjcf/README.md]: Done in https://github.com/gazebosim/gz-mujoco/pull/74/commits/c220e91d2e5279579e07af08b558d15347f891a5

File: sdformat_mjcf/README.md (hunk 2 above)
Comment (azeey) [reply sdformat_mjcf/README.md]: Done in https://github.com/gazebosim/gz-mujoco/pull/74/commits/c220e91d2e5279579e07af08b558d15347f891a5

**Review by scpeters — APPROVED** [review]

**Pull request merged**